cmd/plugins/juju-wait-for/model.go

```diff
@@ -89,16 +89,19 @@ func (c *modelCommand) Init(args []string) (err error) {
 }
 
 func (c *modelCommand) Run(ctx *cmd.Context) error {
-	client, err := c.newWatchAllAPIFunc()
-	if err != nil {
-		return errors.Trace(err)
-	}
-
 	strategy := &Strategy{
-		Client:  client,
-		Timeout: c.timeout,
+		ClientFn: c.newWatchAllAPIFunc,
+		Timeout:  c.timeout,
 	}
-	err = strategy.Run(c.name, c.query, c.waitFor)
+	strategy.Subscribe(func(event EventType) {
+		switch event {
+		case WatchAllStarted:
+			// When a watch has started, we should prime all the local caches,
```

Comment:
Is this something that is intentionally left out of this patch?
This method does nothing...

Reply:
Yes, it's for this PR https://github.com/juju/juju/pull/12196 where I need to evict the cached entities.